dalite_xblock/dalite_xblock.py

```diff
@@ -156,14 +190,74 @@ def student_view(self, context):
         :returns: XBlock HTML fragment
         :rtype: xblock.fragment.Fragment
         """
-        if not self.is_lti_ready:
-            fragment = Fragment()
-            context.update(self._get_context_for_template())
-            fragment.add_content(loader.render_django_template('/templates/xblock_data_not_filled.html', context))
-            return fragment
-        fragment = super(DaliteXBlock, self).student_view(context)
-        fragment.add_javascript(loader.load_unicode('public/js/dalite_xblock.js'))
-        fragment.initialize_js('DaliteXBlock')
+        return self._render_student_view(context, False)
+
+    @contextlib.contextmanager
+    def add_extra_custom_params(self, additional_custom_parameters):
+        """
+        Temporarily adds custom parameters to this xblock.
+
+        These parameters will be sent to the `lti` provider by any calls made during that time.
+
+        :param list additional_custom_parameters: A list of parameters in a 'key=value` format.
+               Eg: ``[u'action=launch-admin']``
+        """
+        current_params = self.custom_parameters
+        try:
+            self.custom_parameters = list(self.custom_parameters) + additional_custom_parameters
+            yield
+        finally:
+            self.custom_parameters = current_params
+
+    @XBlock.handler
+    def lti_launch_handler(self, request, suffix=u''):
+        """
+        Override superclass method.
+
+        This method interprets suffix parameter and translates it to
+        action LTI parameter, which is then passed to dalite.
+        """
+        suffix = unicode(suffix)
+        custom_params = []
+        if suffix == self.ADMIN_URL_SUFFIX:
+            custom_params = [u'action=launch-admin']
+        elif suffix == self.EDIT_QUESTION_SUFFIX:
+            custom_params = [u'action=edit-question']
+
+        with self.add_extra_custom_params(custom_params):
+            return super(DaliteXBlock, self).lti_launch_handler(request)
+
+    def render_button_launching_admin(self, context, form_url_suffix, button_label, id_specifier):
+        """A helper method that renders a button that launches dalite admin in an overlay."""
+        admin_context = dict(context)
+        admin_context.update(self._get_context_for_template())
+        admin_context.update({
+            'has_score': False,
+            'form_url_suffix': form_url_suffix,
+            'dalite_admin_label': button_label,
+            'element_id_specifier': id_specifier
+        })
+
+        return loader.render_django_template("/templates/dalite_xblock_lti_iframe.html", admin_context)
```

Comment:
Addressed in: 73fcc7b2ec8bb925e0cafb82c550fdee0388c015, I ended up copying overlay code from the `xblock-lti-consumer`, other solution would be post upstream PR that adds this functionality, which probably would let to merging this later :(.